Frontend/src/Components/Battery/Battery.js

```diff
@@ -0,0 +1,143 @@
+import {Flex, Center, Box, Spacer, Text, useColorMode} from "@chakra-ui/react";
+import RangeRow from "../Shared/RangeRow";
+import RangeBar from "../Shared/RangeBar";
+import CONSTANTS from "../../data-constants.json";
+import getColor from "../Shared/colors";
+
+export default function Battery(props) {
+    const { colorMode } = useColorMode();
+    const headerBg = getColor("header", colorMode);
+    const borderColor = getColor("border", colorMode);
+
+    const fanEnableText = (props.data?.fan_enable[0] ?? -1.0) ? 'EN' : 'DIS';
+    const fanEnableColor = (props.data?.fan_enable[0] ?? -1.0) ? getColor("greenBg", colorMode) : getColor("redBg", colorMode);
+    const fanEnableTextColor = (props.data?.fan_enable[0] ?? -1.0) ? getColor("txtGreenBg", colorMode) : getColor("txtRedBg", colorMode);
+
+    const chargeEnableText = (props.data?.charge_enable[0] ?? -1.0) ? 'Enable' : 'Disable';
+    const chargeEnableColor = (props.data?.charge_enable[0] ?? -1.0) ? getColor("greenBg", colorMode) : getColor("redBg", colorMode);
+    const chargeEnableTextColor = (props.data?.charge_enable[0] ?? -1.0) ? getColor("txtGreenBg", colorMode) : getColor("txtRedBg", colorMode);
+
+    const fanSpeedMin = ((props.data?.pack_temp[0] ?? -1.0) < (20 + 5 * CONSTANTS.fan_speed.MAX)) ?
+                         Math.floor(((((props.data?.pack_temp[0] ?? -1.0) >= 20) ? props.data?.pack_temp[0] : 15) - 15) / 5) :
+                         CONSTANTS.fan_speed.MAX;
+
+    return (
+        <Flex flex="auto" direction="column" overflowY="scroll">
+            <RangeRow
+                dataTitle="Pack Temp"
+                dataValue={props.data?.pack_temp[0] ?? -1.0}
+                dataConstant={CONSTANTS.pack_temp}
+            />
+            <RangeRow
+                dataTitle="Pack Internal Temp"
+                dataValue={props.data?.pack_internal_temp[0] ?? -1.0}
+                dataConstant={CONSTANTS.pack_internal_temp}
+
+                bg={headerBg}
+            />
+            <Center
+                borderTopWidth={1}
+                borderBottomWidth={1}
+                borderRightWidth={1}
+                borderColor={borderColor}
+                lineHeight='1.2em'
+                height='100%'
+            >
+                <Flex w='95%'>
+                    <Center
+                        border='2px'
+                        borderRadius='0px'
+                        h='75%'
+                        background={fanEnableColor}
+                        marginTop='1%'
+                        marginBottom='1%'
+                        bg={((props.data?.fan_speed[0] ?? -1.0) < fanSpeedMin || (props.data?.fan_speed[0] ?? -1.0) > CONSTANTS.fan_speed.MAX) ?
+        getColor("errorBg", colorMode) : headerBg}
```

Comment:
```suggestion
                        bg={((props.data?.fan_speed[0] ?? -1.0) < fanSpeedMin || (props.data?.fan_speed[0] ?? -1.0) > CONSTANTS.fan_speed.MAX) ?
                            getColor("errorBg", colorMode) : null}
```
**Needs to be moved to the outer `Center` for the row.** Also, to preserve the alternating row colors, the default background color should be `null`. Additionally, nit-picking, but please also make sure the `getColor()` lines up vertically with the opening `{` (or another point that makes sense in the ternary operator)